mrp_lot_number_propagation/wizard/mrp_batch_produce.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from collections import deque
+
+from odoo import api, exceptions, fields, models
+
+
+class MrpBatchProduct(models.TransientModel):
+    _inherit = "mrp.batch.produce"
+
+    is_lot_number_propagated = fields.Boolean(
+        related="production_id.is_lot_number_propagated"
+    )
+    lot_number_propagation_alert_msg = fields.Char(
+        compute="_compute_lot_number_propagation_alert_msg"
+    )
+
+    @api.depends("is_lot_number_propagated")
+    def _compute_lot_number_propagation_alert_msg(self):
+        for wiz in self:
+            msg = ""
+            if wiz.is_lot_number_propagated:
+                propagated_component = (
+                    wiz.production_id._get_propagating_component_move().product_id
+                )
+                msg = self.env._(
+                    f"Lot number must be the same between finished product "
+                    f"{wiz.production_id.product_id.display_name} and component "
+                    f"{propagated_component.display_name}"
+                )
+            wiz.lot_number_propagation_alert_msg = msg
+
+    def _production_text_to_object(self, mark_done=False):
+        self._check_propagated_lot_number()
+        self = self.with_context(lot_propagation=True)
+        return super()._production_text_to_object(mark_done)
+
+    def _check_propagated_lot_number(self):
+        if not self.is_lot_number_propagated:
+            return
+        for production_line in deque(self.production_text.split("\n")):
```

Comment:
I'm not sure you gain that much using `deque` here... You are not consuming the whole list as a queue.
Here a `splitlines` should be enough since you iterate on every element and then `split(sep, 1)` will return only the 1st element. Just a question tho.